### PR TITLE
loop-util: several follow ups for recent changes

### PR DESCRIPTION
**Pull request by yuwata**
Several follow-ups for #24530.

~~Hopefully fixes the portablectl hang issue #24147.~~

### CODE REVIEW

**Comment by yuwata** [comment]
@mrc0mmand Could you test the PR with https://github.com/systemd/systemd-centos-ci/pull/540?

**Comment by yuwata** [comment]
Hm, it seems the issue is not fixed yet...

**Comment by mrc0mmand** [comment]
> Hm, it seems the issue is not fixed yet...

Yeah :( I'm trying to fix the interactive debug stuff, since currently it's pretty much unusable with sanitizers (at least in TEST-29), so I can debug it a bit further.

File: src/shared/loop-util.c [hunk 1]
```diff
@@ -851,7 +851,7 @@ int loop_device_open(
                 .nr = nr,
                 .node = TAKE_PTR(p),
                 .relinquished = true, /* It's not ours, don't try to destroy it when this object is freed */
-                .devno = st.st_dev,
+                .devno = st.st_rdev,
```
Comment (poettering) [inline src/shared/loop-util.c:826]: ouch

**Pull request merged**